public/secure/reset.js

```diff
@@ -0,0 +1,114 @@
+// eslint-disable-next-line max-classes-per-file
+class UnauthorisedError extends Error {}
+class WrongInputError extends Error {}
+
+const urlParams = new URLSearchParams(window.location.search)
+const email = document.getElementById('email')
+const token = document.getElementById('token')
+const form = document.getElementById('password-form')
+const requiredInputs = form.querySelectorAll('[required]')
+const newPassword = document.getElementById('newPassword')
+const newPasswordAgain = document.getElementById('newPasswordAgain')
+
+function performApiRequest(url, data) {
+  return fetch(url, {
+    method: 'POST',
+    body: data,
+    credentials: 'include',
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded',
```

Comment:
Is not this automatic?

Reply:
Looks like the default value is `Content-Type: application/json` However @valerapro is working on accepting JSON. So I will adjust this code once it's done.

Reply:
It's not. I guess it's empty or `text/plain` by default. Since fetch is a low-level API, it should be empty by default.

Reply:
I tried it without and it gave me that header value.